react/src/components/EndpointSelect.tsx

```diff
@@ -24,6 +24,8 @@ const EndpointSelect: React.FC<EndpointSelectProps> = ({
   ...selectProps
 }) => {
   const baiClient = useSuspendedBackendaiClient();
+  const [searchStr, setSearchStr] = useState<string>();
+  const [isSearchPending, startSearchTransition] = useTransition();
   const { baiPaginationOption } = useBAIPaginationOptionState({
     current: 1,
     pageSize: 100,
```

Comment:
Please remove unnecessary code such as pagination hook. You can set `limit` and `skip` directly.  Also 100 is too big. Let change it to 10.